dace/transformation/dataflow/map_distribution.py

```diff
@@ -330,22 +312,15 @@ def can_be_applied(graph: dace.SDFGState,
 
         return True
 
-    @staticmethod
-    def match_to_str(graph: dace.SDFGState, candidate: Dict[pm.PatternNode, int]) -> str:
-        map_entry = graph.node(candidate[ElementWiseArrayOperation2D._map_entry])
-        return map_entry.map.label + ': ' + str(map_entry.map.params)
-
-    def apply(self, sdfg: dace.SDFG):
-        graph = sdfg.nodes()[self.state_id]
-        map_entry = graph.nodes()[self.subgraph[self._map_entry]]
+    def apply(self, graph: dace.SDFGState, sdfg: dace.SDFG):
+        map_entry = self.map_entry
         map_exit = graph.exit_node(map_entry)
 
         sz = dace.symbol('commsize', dtype=dace.int32, integer=True, positive=True)
         Px = dace.symbol('Px', dtype=dace.int32, integer=True, positive=True)
         Py = dace.symbol('Py', dtype=dace.int32, integer=True, positive=True)
 
-        def _prod(sequence):
-            return reduce(lambda a, b: a * b, sequence, 1)
+        from dace.data import _prod
```

Comment:
Maybe we should finally move this to some generic utility submodule? I don't link defining this in multiple files, but importing it from `data` makes no sense.

Reply:
Agreed.